addons/repair/models/repair.py

```diff
@@ -191,8 +191,9 @@ def onchange_partner_id(self):
     @api.onchange('company_id')
     def _onchange_company_id(self):
         if self.company_id:
-            warehouse = self.env['stock.warehouse'].search([('company_id', '=', self.company_id.id)], limit=1)
-            self.location_id = warehouse.lot_stock_id
+            if self.location_id.company_id != self.company_id:
```

Comment:
LGTM. 
But can you put it in the same condition than `self.company_id`? 
=> `if self.company_id and self.location_id.company_id != self.company_id:`

Reply:
That was what I tried first, but in the case where `self.location_id.company_id == self.company_id`, it will go into the else part, and assign the location_id to false, while we just don't want it to change.

Reply:
Ok didn't see the last part. LGTM.